lib/repo/git_repository.rb

```diff
@@ -860,23 +859,12 @@ def make_file(path, file_data, author)
       g.push
     end
 
-    # Make a directory if it's not already present.
-    # If we want the directory creation to have its own commit,
-    # we have to add a dummy file in that directory to do it.
+    # Create and commit an empty directory, if it's not already present.
+    # The dummy file is required so the directory gets committed.
+    # path should be a directory
     def make_directory(path)
-      # Turn "path" into absolute path for repo
-      path = File.expand_path(path, @repos_path)
-
-      # Do nothing if folder already exists
-      return if File.exist?(path)
-
-      # Recursively create parent folders (if doesn't exist)
-      parent_path = File.dirname(path)
-      make_directory(parent_path)
-
-      # Now that the parent folder has been created,
-      # create the current folder
-      FileUtils.mkdir_p(path)
+      gitkeep_filename = File.join(path, '.gitkeep')
+      add_file(gitkeep_filename, '', 'markus')
```

Comment:
I can't figure out how to exercise this code.  It doesn't get called when a new group is created.  I thought it might be called in the rake tasks when we see the database.   The clue is that I can't find a case where a file called .gitkeep is created.  I see .git and .keep.